lib/internal/util/inspect.js

```diff
@@ -228,6 +231,33 @@ function getUserOptions(ctx) {
     getters: ctx.getters,
     ...ctx.userOptions
   };
+
+  // Typically, the target value will be an instance of `Object`. If that is
+  // *not* the case, the object may come from another vm.Context, and we want
+  // to avoid passing it objects from this Context in that case, so we remove
+  // the prototype from the returned object itself + the `stylize()` function,
+  // and remove all other non-primitives, including non-primitive user options.
+  if (isCrossContext) {
+    ObjectSetPrototypeOf(ret, null);
+    for (const key of ObjectKeys(ret)) {
+      if ((typeof ret[key] === 'object' || typeof ret[key] === 'function') &&
+          ret[key] !== null) {
+        delete ret[key];
```

Comment:
I’ve thought about that, but it’s not trivial to determine what the `Object` prototype is in that case.

Reply:
Something like that should work well enough:

```js
let proto = maybeCustom
while(true) {
  const next = Object.getPrototypeOf(proto)
  if (next === null) {
    if (proto === maybeCustom)
      proto = null
    break
  }
  proto = next
}
```

There is no guarantee that it's actually the Object prototype but it should be. It would also be possible to check for the constructor name (which is also not "reliable") but I would just use this.

Reply:
I would prefer to have consistent behavior here rather than to take a guess at the correct prototype.

Reply:
The behavior might actually cause issues inside of the vm context: we return the object prototype in the "regular" context and return a `null` prototype otherwise. The code inside of the VM might not be aware of that and expect an object as prototype and use e.g., `hasOwnProperty`. Using this pattern should work in all cases where the user did not actively manipulate the passed through functions prototype and that is _very_ unlikely.

Reply:
Right, I understand that problem. I still think consistency is more important than the problems that could occur from mis-guessing the prototype.